mlir/lib/Mitigation/Transforms/MitigationMethods/Zne.cpp

```diff
@@ -243,7 +187,168 @@ FlatSymbolRefAttr ZneLowering::getOrInsertFoldedCircuit(Location loc, PatternRew
     // Remove device
     rewriter.create<quantum::DeviceReleaseOp>(loc);
     rewriter.create<func::ReturnOp>(loc, funcFolded);
-    return SymbolRefAttr::get(ctx, fnFoldedName);
+    return SymbolRefAttr::get(rewriter.getContext(), fnFoldedName);
+}
+// In *.cpp module only, to keep extraneous headers out of *.hpp
+FlatSymbolRefAttr randomLocalFolding(PatternRewriter &rewriter, std::string fnFoldedName,
+                                     func::FuncOp fnFoldedOp, Value c0, Value c1)
+{
+    // TODO: Implement.
+
+    // Can't throw, because disabled by compilation.
+    // throw std::logic_error("Random local folding not implemented!");
+
+    return FlatSymbolRefAttr();
+}
+// In *.cpp module only, to keep extraneous headers out of *.hpp
+FlatSymbolRefAttr allLocalFolding(PatternRewriter &rewriter, std::string fnFoldedName,
+                                  func::FuncOp fnFoldedOp, Value c0, Value c1)
+{
+    // Type qregType = quantum::QuregType::get(rewriter.getContext());
+
+    int64_t sizeArgs = fnFoldedOp.getArguments().size();
+    Value size = fnFoldedOp.getArgument(sizeArgs - 1);
+
+    // Walk through the operations in fnWithMeasurementsOp
+    fnFoldedOp.walk([&](quantum::QuantumGate op) {
+        rewriter.setInsertionPoint(op);
+        auto loc = op->getLoc();
+        const std::vector<Value> opQubitArgs = op.getQubitOperands();
+
+        // Insert a for loop immediately before each quantum::QuantumGate
+        const auto forVal =
+            rewriter
+                .create<scf::ForOp>(
+                    loc, c0, size, c1, /*iterArgsInit=*/opQubitArgs,
+                    [&](OpBuilder &builder, Location loc, Value i, ValueRange iterArgs) {
+                        // Set insertion point within the loop
+                        builder.setInsertionPointToEnd(builder.getBlock());
```

Comment:
I might be wrong but I think that you can remove this line?

Reply:
Done.